internal/controllers/machinehealthcheck/machinehealthcheck_targets_test.go

```diff
@@ -426,6 +426,13 @@ func TestHealthCheckTargets(t *testing.T) {
 			expectedNeedsRemediationCondition: []clusterv1.Condition{machineFailureMsgCondition},
 			expectedNextCheckTimes:            []time.Duration{},
 		},
+		{
+			desc:                              "health checl without conditions",
```

Comment:
The healthCheckTarget in Targets: here would need to be something like the one in `nodeGoneAway` but with a MHC with no conditions.

Reply:
I used this, but test fails.

```
	// Create a test MHC without conditions
	testMHCEmptyConditions := &clusterv1.MachineHealthCheck{
		ObjectMeta: metav1.ObjectMeta{
			Name:      "test-mhc",
			Namespace: namespace,
		},
		Spec: clusterv1.MachineHealthCheckSpec{
			Selector: metav1.LabelSelector{
				MatchLabels: mhcSelector,
			},
			ClusterName: clusterName,
		},
	}


	// Target for when the Node has been seen, but has now gone
	// using MHC without unhealthyConditions
	nodeGoneAwayEmptyConditions := healthCheckTarget{
		Cluster:     cluster,
		MHC:         testMHCEmptyConditions,
		Machine:     testMachine.DeepCopy(),
		Node:        &corev1.Node{},
		nodeMissing: true,
	}

...
		{
			desc:                              "health check with empty unhealthy conditions",
			targets:                           []healthCheckTarget{nodeGoneAwayEmptyConditions},
			expectedHealthy:                   []healthCheckTarget{},
			expectedNeedsRemediationCondition: []clusterv1.Condition{nodeGoneAwayCondition},
			expectedNextCheckTimes:            []time.Duration{},
		},

```


```
E1205 15:40:01.716195   74781 controller.go:329]  "msg"="Reconciler error" "error"="machinehealthchecks.cluster.x-k8s.io \"test-mhc-qmbzd\" not found" "MachineHealthCheck"={"name":"test-mhc-qmbzd","namespace":"test-mhc-7pw4k"} "controller"="machinehealthcheck" "controllerGroup"="cluster.x-k8s.io" "controllerKind"="MachineHealthCheck" "name"="test-mhc-qmbzd" "namespace"="test-mhc-7pw4k" "reconcileID"="505b8aad-776e-40eb-90ed-ad12c4b3a071"
I1205 15:40:01.721890   74781 machinehealthcheck_targets.go:209]  "msg"="skipping remediation" "Machine"={"name":"machine5","namespace":"test-mhc"} "reason"="machine has \"cluster.x-k8s.io/skip-remediation\" annotation"
I1205 15:40:01.721902   74781 machinehealthcheck_targets.go:209]  "msg"="skipping remediation" "Machine"={"name":"machine6","namespace":"test-mhc"} "reason"="machine has \"cluster.x-k8s.io/paused\" annotation"
--- FAIL: TestHealthCheckTargets (0.00s)
    --- FAIL: TestHealthCheckTargets/health_check_with_empty_unhealthy_conditions (0.00s)
        machinehealthcheck_targets_test.go:505: 
            Expected
                <[]machinehealthcheck.healthCheckTarget | len:1, cap:1>: [
                    {

```

Reply:
This is now done, can you check?